File: Assignment_2/arrayPairSum.java
package Assignment_2;

/* Ques 1 : Given an integer array nums of 2n integers, group these integers into n pairs (a1, b1), (a2, b2),..., (an, bn) such that the sum of min(ai, bi) for all i is maximized. Return the maximized sum.*/

import java.util.Arrays;

class Solution {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int sum = 0 ;
        for(int j = 0 ; j<nums.length ; j = j+2){
            sum = sum + nums[j];
        }
        return sum;
    }
}

/*Note: This is a leetcode solution that I have done and successfully submitted */

File: Assignment_4/arrayPairSum.js
/* Ques 4 : Given an integer array nums of 2n integers, group these integers into n pairs (a1, b1), (a2, b2), ..., (an, bn) such that the sum of min(ai, bi) for all i is maximized. Return the maximized sum.*/

/**
 * @param {number[]} nums
 * @return {number}
 */

var arrayPairSum = function (nums) {
  let sum = 0;

  function compareNumbers(a, b) {
    return a - b;
  }
//   sort number array 
  nums.sort(compareNumbers);

  for (let i = 0; i < nums.length; i = i + 2) {
    sum = sum + nums[i];
  }
  return sum;
};

// Note : Done problem in js and successfully submit on leetcode .
